File: Codes/3206-alternating-groups-i.cpp
/*
 * @Author: LetMeFly
 * @Date: 2024-11-26 23:18:14
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2024-11-26 23:18:20
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    int numberOfAlternatingGroups(vector<int>& colors) {
        int ans = 0;
        for (int i = 0; i < colors.size(); i++) {
            if (colors[i] != colors[(i + 1) % colors.size()] && colors[i] == colors[(i + 2) % colors.size()]) {
                ans++;
            }
        }
        return ans;
    }
};

File: Codes/3206-alternating-groups-i.py
'''
Author: LetMeFly
Date: 2024-11-26 23:20:18
LastEditors: LetMeFly.xyz
LastEditTime: 2024-11-26 23:20:28
'''
from typing import List

class Solution:
    def numberOfAlternatingGroups(self, colors: List[int]) -> int:
        return sum(colors[i] != colors[(i + 1) % len(colors)] != colors[(i + 2) % len(colors)] for i in range(len(colors)))